node/engine/messageservice/p2p-message-service/dht-record.go

```diff
@@ -0,0 +1,85 @@
+package p2pms
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/libp2p/go-libp2p/core/peer"
+)
+
+const (
+	DHT_RECORD_PREFIX = "/scaddr/"
+)
+
+type stateChannelAddrToPeerIDValidator struct{}
+
+// DhtRecord represents the data stored in the DHT record
+type DhtRecord struct {
+	Data      DhtData `json:"data"`
+	Signature []byte  `json:"signature"`
+}
+
+type DhtData struct {
+	SCAddr    string `json:"scaddr"` // state channel address
```

Comment:
Is it worth using these struct tags to lowercase the JSON ouput? Or could we just omit them to simplify things?

Reply:
@lalexgap are you suggesting to omit the `json` tags so that the struct looks like this?
```
type dhtData struct {
	SCAddr    string // state channel address
	PeerID    string
	Timestamp int64  // Unix timestamp (seconds since January 1, 1970)
}
```

Reply:
@bitwiseguy Yup! The fields will still be outputted to JSON (but as `ScAddr` instead of `scaddr`) and we save same text.

Reply:
@kerzhner I wanted to add the `SCAddr` to that struct so that it is included in the `Signature`. So the signature directly ties the `PeerID` to the `SCAddr`. But maybe that is unnecessary?

Reply:
That's a good point, and we should keep `SCAddr` in the struct. If we do not include the `SCAddr`, then the record and signature can be reused by any peer. A malicious peer can then add the record and signature pair for any state channel address key.

Reply:
Added a comparison of the record's key to the `SCAddr` value